cmd/influx/cli/cli.go

```diff
@@ -729,9 +758,30 @@ func (c *CommandLine) ExecuteQuery(query string) error {
 		}
 		query = pq.String()
 	}
-	response, err := c.Client.Query(c.query(query))
+	return query, nil
+}
+
+// ExecuteQueryWithCancel runs any query statement. It is cancellable by the user.
+func (c *CommandLine) ExecuteQueryWithCancel(query string) error {
```

Comment:
I think you can do this without necessarily having a separate control flow for `ExecuteQuery` and `ExecuteQueryCancel`. The `ExecuteQuery` function has access to the signals being received by the program already so you don't need two methods. All you really need is something like this.

```
ctx := context.Background()
if !c.IgnoreSignals {
    done := make(chan struct{})
    defer close(done)

    var cancel func()
    ctx, cancel = context.WithCancel(ctx)
    go func() {
        // wait for either a signal to be sent or the done channel to have been closed
        select {
        case <-done:
        case <-c.osSignals:
            cancel()
        }
    }()
}
// actually execute the query here with the context
```

Reply:
No, I can't do that, because of the way that contexts work. I need on ExecuteQueryCancel to call clientDoQuery to be able to a) catch cancels and b) get the response on the closure. This is the standard way to use contexts as described here: https://blog.golang.org/context 
Also, the main loop has access to the signals, ExecuteQuery does not, that's why I has to find a way to make them accessible and that's why I'm using context.